runtime/compiler/trj9/env/VMJ9.cpp

```diff
@@ -6518,6 +6518,56 @@ TR_J9VMBase::getResolvedMethods(TR_Memory * trMemory, TR_OpaqueClassBlock * clas
       }
    }
 
+
+
+/**
+ * @brief Create a TR_ResolvedMethod given a class, method name and signature
+ *
+ * The function will scan all methods from the given class until a match for
+ * the method name and signature is found
+ *
+ * @param trMemory     Pointer to TR_Memory object used for memory allocation
+ * @param classPointer The j9class that is searched for method name/signature
+ * @param methodName   Null terminated string denoting the method name we are looking for
+ * @param signature    Null terminated string denoting the signature of the method we want
+ * @return             A pointer to a TR_ResolvedMethod for the indicated j9method, or null if not found
+ */
+TR_ResolvedMethod *
+TR_J9VMBase::getResolvedMethodForNameAndSignature(TR_Memory * trMemory, TR_OpaqueClassBlock * classPointer,
+                                                  const char* methodName, const char *signature)
+   {
+   TR::VMAccessCriticalSection vmCS(this); // Prevent HCR
```

Comment:
What does "Prevent HCR" mean here? This is new to me. Could you explain?

Reply:
HCR=Hot code replacement. This acronym is used in many places throughout the code.
I had a question whether acquiring VM access is actually needed in that piece of code. We know that class unloading cannot happen because the JIT always holds the class unload monitor. However, without VM access, HCR could happen while we iterate over methods in a class and GAC says this could lead to methods being seen twice or not a lot. Moreover, under FSD methods can be added or replaced. Thus, for safety we need to prevent HCR while iterating over methods.